### DISCUSSION TITLE
How to convert a TIF file with a clipping mask to PNG?

### DISCUSSION THREAD

**Discussion by zxc155226522** [ImageMagick/ImageMagick#6529] [Help]
### ImageMagick version

7.1.1-6

### Operating system, version and so on

win10

### Description

![image](https://github.com/ImageMagick/ImageMagick/assets/129932013/0b91384f-21af-439f-835a-a445a3a39d4a)

[230101-1.zip](https://github.com/ImageMagick/ImageMagick/files/12226724/230101-1.zip)

**Comment by snibgo** [comment]
```
magick "230101-1 (2).tif" info:
230101-1 (2).tif[0] TIFF 1193x1606 1193x1606+0+0 8-bit ColorSeparation CMYK 32.5541MiB 0.016u 0:00.014
230101-1 (2).tif[1] TIFF 1193x1606 1193x1606+0+0 8-bit CMYK 32.5541MiB 0.016u 0:00.013
230101-1 (2).tif[2] TIFF 1181x1594 1181x1594+6+6 8-bit CMYK 32.5541MiB 0.000u 0:00.010
230101-1 (2).tif[3] TIFF 1193x1606 1193x1606+0+0 8-bit CMYK 32.5541MiB 0.000u 0:00.005
```
The file contains four images. Do you want all of them, or just one?

The images are all CMYK, without embedded ICC profiles. PNG can't record CMYK, so the pixels must be converted to use RGB channels. A simple method to convert just the first image is:
```
magick "230101-1 (2).tif[0]" out.png
```

**Reply by zxc155226522** [reply]
Translation: It is best to convert to the same effect as seen in tif. Otherwise, if it is done in batches, each one needs to be selected individually.

**Comment by snibgo** [comment]
If you want to convert all the files in a directory, either put that command in a shell loop, or use `magick mogrify`:
```
magick mogrify -format png *.tif[0]
```

**Reply by zxc155226522** [reply]
But there is an alpha channel w1, and if transparency is not saved when saving in Photoshop, the background will be white.
[230101-1.zip](https://github.com/ImageMagick/ImageMagick/files/12235637/230101-1.zip)
